src/adapter/src/catalog/consistency.rs

```diff
@@ -348,6 +358,143 @@ impl CatalogState {
             Err(dependency_inconsistencies)
         }
     }
+
```

Comment:
You could probably pull this whole commit out into a separate PR.

Reply:
I pulled this our into https://github.com/MaterializeInc/materialize/pull/22411